front/planning.php

```diff
@@ -128,8 +128,9 @@
             }
          }
 
-         if ($ismine || $canview) {
+         if (($ismine || $canview) && Session::initForUser($_GET["uID"])) {
```

Comment:
Could the solution be to replace `if ($user->getFromDBByToken($_GET['token'])) {` by something like `if ($auth->authWithToken($_GET['token'], 'personal_token')) {`, where `Auth::authWithToken()` would be a new method that would do something like:

```php
public static function authWithToken(string $token, string $token_type): bool {
   $user = new User();
   if ($user->getFromDBByToken($token, $token_type)) {
      $auth = new Auth();
      $auth->auth_succeded = true;
      Session::init($auth);
      return true;
   }
   return false;
}
```

Reply:
or maybe just replace the call to `loadMinimalSession` with `Session::initForUser` to get the real session directly ?

Reply:
If it works, yes. My point was to know if it is possible to init the session for the user related to given token instead of opening the session for another user.

Reply:
Ok, I thought the token and ids were always from the same user but it's not the case.
I'll change the function to authWithToken then (and replace loadMinimalSession since it would destroy the session).

Reply:
Should be all good now